client/signup/config/flows.js

```diff
@@ -197,7 +198,13 @@ const Flows = {
 		}
 
 		if ( isUserLoggedIn ) {
-			flow = removeUserStepFromFlow( flow );
+			const urlParams = new URLSearchParams( window.location.search );
+			const param = urlParams.get( 'user_completed' );
+			// Remove the user step unless the user has just completed the step
+			// and then clicked the back button.
+			if ( ! param && ! detectHistoryNavigation.loadedViaHistory() ) {
+				flow = removeUserStepFromFlow( flow );
```

Comment:
Can we confirm that this won't break something for an existing signup flow?

Reply:
`flow = removeUserStepFromFlow( flow );` was there before. 
It used to be always applied. To test that it still works as expected. 
I create the user and then just visit `/start/user` it should then redirect me to => `/start/domains` 

I am not sure how else to test this?

Reply:
There is also flows like `p2` and `p2-new` where the user step isn't the first step. I briefly tried them and the step exclusion still seems to work as expected.